LXC Support

fpletz opened the pull request:
@elseym

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/mayflower/baseboxes/1)

<!-- Reviewable:end -->

ironpinguin committed:
initial add lxc box packer build.

ironpinguin committed:
ignore post-processor vagrant for lxc because it not work with lxc.

ironpinguin committed:
no use user variables in post-processor for vagrant

This is neede because in the master branch of packer is a bug.
To create vagrant box we use a branch from master.

ironpinguin committed:
update short description to build lxc box.

ironpinguin committed:
Fix language setting for lxc and add man pages

fpletz committed:
Replace Ubuntu archive mirror for every country

ironpinguin committed:
Update the lxc.md with more detail prepare description.

ironpinguin committed:
create shell script to build packer.

ironpinguin committed:
add script

globin commented on lxc/ubuntu:
```diff
@@ -0,0 +1,66 @@
+# When using LXC with apparmor, the container will be confined by default.
+# If you wish for it to instead run unconfined, copy the following line
+# (uncommented) to the container's configuration file.
+#lxc.aa_profile = unconfined
```
could you uncomment this, as this breaks on newer ubuntu and NixOS currently, this can be fixed per user but I think the problems outweigh the gains. @fpletz ?

fpletz replied:
Ack, we don't really need Apparmor because the use case for our baseboxes is development and not production.

globin commented on templates/trusty64-puppet3.json:
```diff
@@ -136,16 +151,27 @@
   "post-processors": [
     {
-      "output": "{{user `output_directory`}}/trusty64-puppet3-{{user `box_version`}}-{{.Provider}}.box",
+      "output": "builds/trusty64-puppet3-0.7.1-{{.Provider}}.box",
```
Is the bug preventing us from using this still present?

globin commented:
Apart from my comments is there anything preventing us from merging this?

ironpinguin committed:
Fix some errors in build script and description.

fpletz commented:
LGTM except your comments.

globin committed:
Use the unconfined apparmor profile

This is only supposed to be used in dev environments and not in
production so support recent ubuntu and NixOS
